### PR TITLE
BackupScheduleSetting: enable fetching and updating schedule time

### PR DESCRIPTION
**Pull request by Initsogar**
Related to https://github.com/Automattic/jetpack-backup-team/issues/669

## Proposed Changes

Allow users to update their daily backup time schedule. This is still under the `jetpack/backup-schedule-setting` until we decide to release it.

* Add `useScheduledTimeQuery` to handle scheduled time fetch. This one will be reused to display the scheduled time in the `Latest backup` section as part of https://github.com/Automattic/jetpack-backup-team/issues/700
* Add `useScheduledTimeMutation` to handle scheduled time update
* Enable `jetpack/backup-schedule-setting` on Jetpack Cloud development and horizon (live branches) environments

## Demo

https://github.com/user-attachments/assets/c5b08c2b-3697-40b7-8c36-c9f0a7d2ebff

## Why are these changes being made?

https://github.com/Automattic/jetpack-backup-team/issues/669

## Testing Instructions

* Spin up a Jetpack Cloud live branch
* Pick a site with Jetpack VaultPress Backup plan
* Navigate to the Settings page
* You should see the `Daily backup time schedule` setting at the bottom of the page
* If you never updated your schedule time, you should see a default one with the `Default time` message
* Pick another time slot on the dropdown.
  * The dropdown should be in your local time (the one configured on your site). So if you click 2:00pm (your local time), it will be stored in UTC.
  * When you click it, it will trigger the update (you should see the dropdown being disabled for a bit)
  * Once it completes, you should see a  success notice with the message `Daily backup time successfully changed.`
  * The dropdown should be updated with the new value you selected.
  * The message below the dropdown should be changed to `Time set by <user>. UTC: X:XX - X:XX XX`.
    * <user> will be the name of the user who performed the action
    * `UTC: X:XX - X:XX XX` will be the UTC time of the schedule time saved.
## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/components/jetpack/backup-schedule-setting/index.tsx [hunk 1]
```diff
@@ -1,39 +1,118 @@
...
+	const getScheduleInfoMessage = (): TranslateResult => {
+		const hour = data?.scheduledHour || 0;
+		const range = convertHourToRange( hour );
+
+		if ( ! data || ! data.scheduledBy ) {
+			return `${ translate( 'Default time' ) }. UTC: ${ range }`;
+		}
+		return `${ translate( 'Time set by %(scheduledBy)s', {
+			args: { scheduledBy: data.scheduledBy },
+		} ) }. UTC: ${ range }`;
```
Comment (mavegaf) [inline client/components/jetpack/backup-schedule-setting/index.tsx:102]: In the case of UTC, there is no AM/PM, it uses 24-hour clock

Comment (Initsogar) [reply client/components/jetpack/backup-schedule-setting/index.tsx:102]: Oh, TIL. Fixing it!

**Review by mavegaf — APPROVED** [review]
👍

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16927149).

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16927149).

**Pull request merged**